native/acter/src/api/verification.rs

```diff
@@ -114,10 +114,65 @@ impl VerificationEvent {
         self.emojis.clone_from(&emojis);
     }
 
-    pub fn get_emojis(&self) -> Vec<VerificationEmoji> {
+    // when this device triggered verification of other device, it can get emojis from SAS state
+    pub fn emojis(&self) -> Vec<VerificationEmoji> {
         self.emojis.clone()
     }
 
+    // when other device triggered verification of this device, it can get emojis from remote server
+    pub async fn get_emojis(&self) -> Result<Vec<VerificationEmoji>> {
+        let client = self.client.clone();
+        let sender = self.sender.clone();
+        let event_id = self.event_id.clone();
+        let txn_id = self.txn_id.clone();
+        RUNTIME
+            .spawn(async move {
+                if let Some(event_id) = event_id {
+                    if let Some(Verification::SasV1(sas)) = client
+                        .encryption()
+                        .get_verification(&sender, event_id.as_str())
+                        .await
+                    {
+                        if let Some(items) = sas.emoji() {
+                            let sequence = items
+                                .iter()
+                                .map(|e| VerificationEmoji {
+                                    symbol: e.symbol.chars().next().unwrap() as u32, // first char in string
+                                    description: e.description.to_string(),
+                                })
+                                .collect::<Vec<VerificationEmoji>>();
+                            return Ok(sequence);
+                        } else {
+                            return Ok(vec![]);
+                        }
```

Comment:
Is this really a proper case? Shouldn't this raise an error that we didn't receive any emojis?

```suggestion
                        let Some(items) = sas.emoji() else {
                            bail!('No emojis found. Aborted.');
                        };
                        let sequence = items
                            .iter()
                            .map(|e| VerificationEmoji {
                                symbol: e.symbol.chars().next().unwrap() as u32, // first char in string
                                description: e.description.to_string(),
                            })
                            .collect::<Vec<VerificationEmoji>>();
                        return Ok(sequence);
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/886/commits/d3b14ed0add97f31c0c55f897925f668dcccce6b